freqtrade/strategy/strategy_wrapper.py

```diff
@@ -23,7 +23,16 @@ def wrapper(*args, **kwargs):
         try:
             if 'trade' in kwargs:
                 # Protect accidental modifications from within the strategy
-                kwargs['trade'] = deepcopy(kwargs['trade'])
+                trade = kwargs['trade']
+                if isinstance(trade, Trade):
+                    if trade in Trade.query.session.dirty:
+                        Trade.commit()
```

Comment:
This will add random commits at random points in the process - i'm not entirely sure that this won't have unintended side-effects.